client/dispatcher.go

```diff
@@ -50,15 +52,34 @@ func createUploaders(targets []string) []uploader {
 	return uploaders
 }
 
-func sendMsgToPublicUploaders(msg []byte, topic string) {
-	sendMsgToUploaders(msg, topic, dis.publicUploaders)
-	sendMsgToUploaders(msg, topic, dis.privateUploaders)
-	sendMsgToWebSockets(msg, topic)
+func sendMsgToPublicUploaders(upload interface{}, topic string, state *albionState) {
+	data, err := json.Marshal(upload)
+	if err != nil {
+		log.Errorf("Error while marshalling payload for %v: %v", err, topic)
+		return
+	}
+
+	sendMsgToUploaders(data, topic, dis.publicUploaders)
+	sendMsgToUploaders(data, topic, dis.privateUploaders)
+	sendMsgToWebSockets(data, topic)
 }
 
-func sendMsgToPrivateUploaders(msg []byte, topic string) {
-	sendMsgToUploaders(msg, topic, dis.privateUploaders)
-	sendMsgToWebSockets(msg, topic)
+func sendMsgToPrivateUploaders(upload lib.PersonalizedUpload, topic string, state *albionState) {
+	if state.CharacterName == "" || state.CharacterId == "" {
+		log.Error("The player name or id has not yet been set. Please transition zones so the name can be identified.")
```

Comment:
Does zone transition actually give us that? I was under the impression the Join operation was on character selection or initial connection.

Reply:
I have to re-check my logs but i'm pretty sure you get the Join operation everytime you switch zones. It's the same operation that has been used for the location id before. The text is also copy-paste

Reply:
We primarily used the GetGameServerByCluster. The Join was added later to get it on initial login so you didn't have to transition zones after first login.

Reply:
Yup, you are correct. Hmm ok then i'm not sure how to handle the missing name or id. Seems like the only option at the moment would be to restart. I do know that there are some other events that include the name and/or id and that get sent later, might be an option to parse them as well.

Reply:
Yup, I think its also OK to tell people that they need to start our client before starting the albion client.

One of the things we need to figure out is how to notify people on windows they need to do things.

See newly created #134